lib/jack_tokenizer.rb

```diff
@@ -7,7 +7,7 @@ def initialize(io)
   KEYWORD_TOKENS = %w[class method function constructor int boolean char void var static field let do if else while return true false null this]
 
   def has_more_tokens?
-    if (match = @input.match(%r{([ \n]+|//.*$|/\*(.|\n)*\*/)+}, @index)) && match.begin(0) == @index
+    if (match = @input.match(%r{([ \n\r\t]+|//.*$|/\*(.|\n)*\*/)+}, @index)) && match.begin(0) == @index
```

Comment:
Note we would’ve got this for free if we’d used the more general `\s` instead of the more explicit `[ \n]`.